### PR TITLE
AO3-6009 Policy & Abuse link in first login banner

### PR DESCRIPTION
**Pull request by tlee911**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6009

## Purpose

Adds a text and link to the first login message telling users about policy and abuse and what they can do if they experience harassment.  All user-facing strings have been added to localization config to support future translations.

## Testing Instructions

Feature file for the login banner updated to test this.

For manual testing:
- Create new account or re-enable the first login message
- View the first login banner
- Verify the text and link are displayed correctly
- Verity the link takes user to policy and abuse page

## References

Same reference linked in JIRA:
https://citizensandtech.org/2020/06/reducing-harassment-impacts-in-feminism-online/

## Credit

tlee911 (they/them)

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -2,16 +2,17 @@
   <!-- pseudoflash because putting it in a real one was making for some ugly code -->
   <div class="flash notice" id="first-login-help-banner">
     <p>
-      Hi! It looks like you've just logged into the archive for the first time. 
-      <%= link_to_modal ts("Learn some tips and tricks"), :for => first_login_help_path, :title => "First login help" %> 
-      or dismiss this message permanently.
+      <%= t(".login_banner.welcome_text", help_link: link_to_modal(t(".login_banner.help_link"), :for => first_login_help_path, :title => t(".login_banner.help_title"))).html_safe %>
```
Comment (sarken) [inline app/views/users/show.html.erb:5]: Would you mind updating the syntax where possible as long as you're here, so it's `for: first_login_help_path`, etc?

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -2,16 +2,17 @@
...
-      <%= form_tag end_first_login_user_path(current_user), :method => :post, :remote => true do %>
+    <p>
+      <%= t(".login_banner.abuse_text", abuse_link: link_to(t(".login_banner.abuse_link"), "/abuse_reports/new")).html_safe %>
+    </p>
+    <%= form_tag end_first_login_user_path(current_user), :method => :post, :remote => true do %>
     <p class="submit actions">
-      <%= submit_tag ts("Dismiss permanently") %>
-      <%= link_to "x", nil, :remote => true, :id => "hide-first-login-help", :title => "Hide first login help banner" %>
+      <%= submit_tag t(".login_banner.dismiss") %>
+      <%= link_to "x", nil, :remote => true, :id => "hide-first-login-help", :title => t(".login_banner.hide") %>
```
Comment (sarken) [inline app/views/users/show.html.erb:13]: One more annoying "as long as you're here..." thing: can you make sure this is [the times symbol like we use in admin banners](https://github.com/otwcode/otwarchive/blob/f057b6b9f85ccdd26dcff2aca2efa17cf7f38e26/app/views/layouts/_banner.html.erb#L18) and not a letter `x`? (I _think_ it's currently the letter, but it's hard to tell in in certain fonts unless you see the two things side by side.)

**Review by sarken — CHANGES_REQUESTED** [review]
Thanks for this! I'm afraid I've left you an annoying number of comments, since the existing code is a bit old and could use some updating to bring it more in line with what we currently do -- sorry!

File: features/other_a/banner_login.feature [hunk 3]
```diff
@@ -6,14 +6,23 @@ Feature: First login help banner
   Given I am logged in as "newname"
   When I am on newname's user page
   Then I should see the first login banner
-  
+    And I should see "Hi! It looks like you've just logged into the archive for the first time."
```
[the remark this answers is not in the document]
Comment (redsummernight) [reply features/other_a/banner_login.feature:9]: I think you can also combine "Scenario: New user sees the banner" with the one you're adding for the Policy & Abuse link. They're similar enough it's not worth running the test setup twice.

File: app/views/users/show.html.erb (hunk 2 above)
Comment (tlee911) [reply app/views/users/show.html.erb]: TIL "&times;" was a thing.  This was totally the letter.

File: features/other_a/banner_login.feature (hunk 3 above)
[the remark this answers is not in the document]
Comment (tlee911) [reply features/other_a/banner_login.feature]: I've combined the two tests as suggested.  Is it worth also combining with the [link to modal test](https://github.com/otwcode/otwarchive/blob/master/features/other_a/banner_login.feature#L10-L15)?

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -2,16 +2,17 @@
...
-      <%= submit_tag ts("Dismiss permanently") %>
-      <%= link_to "x", nil, :remote => true, :id => "hide-first-login-help", :title => "Hide first login help banner" %>
+    <p>
+      <%= t(".login_banner.abuse_text", abuse_link: link_to(t(".login_banner.abuse_link"), new_abuse_report_path)).html_safe %>
     </p>
-      <% end %>
+    <%= form_tag end_first_login_user_path(current_user), method: :post, remote: true do %>
+      <p class="submit actions">
+        <%= submit_tag t(".login_banner.dismiss") %>
+        <%= link_to "&times;".html_safe, nil, remote: true, id: "hide-first-login-help", title: t(".login_banner.hide") %>
```
Comment (sarken) [inline app/views/users/show.html.erb:13]: It looks like lines 42 and 49 of features/other_a/banner_login.feature need to be updated to use the new character.

File: features/other_a/banner_login.feature (hunk 3 above)
[the remark this answers is not in the document]
Comment (sarken) [reply features/other_a/banner_login.feature]: I'd lean toward leaving them separate -- checking that the popup is functioning feels different than checking the content of the banner.

**Review by sarken — APPROVED** [review]

**Pull request merged**